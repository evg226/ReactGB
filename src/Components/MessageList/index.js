import "./style.scss";

import { Paper, List, Typography, ListItem, ListItemAvatar, Avatar, ListItemText } from '@material-ui/core';

function MessageList({messageList}) {
    return (
        <Paper variant="outlined">
            <List className="message-list">{messageList.map(message => < Message key={message.id} mes={message} />)} </List>
        </Paper>
    )
}

export const Message=(props) =>{
    return (
            <ListItem alignItems="flex-start" className={props.mes.author==="Robot"?"message-item message-item_right":"message-item"}>
                <ListItemAvatar>
                    <Avatar alt={props.mes.author} src="img/1.jpg" />
                </ListItemAvatar>
                <ListItemText
                    primary={<Typography component="div" variant="body1" color="textPrimary">{props.mes.author}</Typography>}
                    secondary={<Typography component="div" variant="body2" color="textPrimary">{props.mes.text}</Typography>}
                />
            </ListItem>
    )
}

export default MessageList;