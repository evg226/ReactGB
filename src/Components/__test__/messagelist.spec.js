import { render } from "@testing-library/react";
import MessageList, {Message}  from "../MessageList";


describe("Тестирование компонента-контейнера MessageList и презентационного - message", () => {
    it("MessageList c параметрами", () => {
        const component = render(<MessageList messageList={[{ id: 1, Author: "Me", text: "Hello" }]} />);
        const message = component.getByText("Hello");
        expect(message).toBeInTheDocument(); //проверяет присутствие сообщения в дочернем элементе
    });
    it("MessageList c пустыми параметрами", () => {
        const component = render(<MessageList messageList={[]} />);
        expect(component).toMatchSnapshot(); 
    });
    it("Message c параметрами", () => {
        const component = render(<Message mes={{ id: 1, author: "Me", text: "Hello" }}/>);
        expect(component).toMatchSnapshot();
    });
})