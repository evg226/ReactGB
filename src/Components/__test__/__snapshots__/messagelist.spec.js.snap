// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента-контейнера MessageList и презентационного - message Message c параметрами 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular"
          >
            <img
              alt="Me"
              class="MuiAvatar-img"
              src="img/1.jpg"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            Me
          </div>
          <div
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
          >
            Hello
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    >
      <div
        class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular"
        >
          <img
            alt="Me"
            class="MuiAvatar-img"
            src="img/1.jpg"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
        >
          Me
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Hello
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Тестирование компонента-контейнера MessageList и презентационного - message MessageList c пустыми параметрами 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
      >
        <ul
          class="MuiList-root message-list MuiList-padding"
        >
           
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
    >
      <ul
        class="MuiList-root message-list MuiList-padding"
      >
         
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Тестирование компонента-контейнера MessageList и презентационного message Message c параметрами 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular"
          >
            <img
              alt="Me"
              class="MuiAvatar-img"
              src="img/1.jpg"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            Me
          </div>
          <div
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
          >
            Hello
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    >
      <div
        class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular"
        >
          <img
            alt="Me"
            class="MuiAvatar-img"
            src="img/1.jpg"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
        >
          Me
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Hello
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Тестирование компонента-контейнера MessageList и презентационного message Message c пустыми параметрами 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular"
          >
            <img
              alt="Me"
              class="MuiAvatar-img"
              src="img/1.jpg"
            />
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            Me
          </div>
          <div
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
          >
            Hello
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    >
      <div
        class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular"
        >
          <img
            alt="Me"
            class="MuiAvatar-img"
            src="img/1.jpg"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
        >
          Me
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Hello
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Тестирование компонента-контейнера MessageList и презентационного message MessageList c параметрами 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
      >
        <ul
          class="MuiList-root message-list MuiList-padding"
        >
          <li
            class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
          >
            <div
              class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular"
              >
                <img
                  class="MuiAvatar-img"
                  src="img/1.jpg"
                />
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <div
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
              />
              <div
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
              >
                Hello
              </div>
            </div>
          </li>
           
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
    >
      <ul
        class="MuiList-root message-list MuiList-padding"
      >
        <li
          class="MuiListItem-root message-item MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular"
            >
              <img
                class="MuiAvatar-img"
                src="img/1.jpg"
              />
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <div
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
            />
            <div
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
            >
              Hello
            </div>
          </div>
        </li>
         
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Тестирование компонента-контейнера MessageList и презентационного message MessageList c пустыми параметрами 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
      >
        <ul
          class="MuiList-root message-list MuiList-padding"
        >
           
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
    >
      <ul
        class="MuiList-root message-list MuiList-padding"
      >
         
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
